app/controllers/datasets/frequencies_controller.rb

```diff
@@ -11,7 +11,14 @@ def edit
 
   def create
     @dataset = current_dataset
-    @dataset.frequency = params.require(:dataset).permit(:frequency)[:frequency]
+
+    begin
+      @dataset.frequency = params.require(:dataset).permit(:frequency)[:frequency]
+    rescue ActionController::ParameterMissing
+      @dataset.errors.add(:frequency, 'No frequency selected')
+      render 'new'
+      return
+    end
```

Comment:
I think maybe you can use `rescue_from with :handler` http://api.rubyonrails.org/classes/ActiveSupport/Rescuable/ClassMethods.html